ts/WoltLabSuite/Core/Ui/Feed/Dialog.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Shows a dialog with an anonymous and personalized link for RSS feeds with access tokens.
+ *
+ * @author  Matthias Schmidt
+ * @copyright  2001-2021 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Feed/Dialog
+ */
+
+import UiDialog from "../Dialog";
+import * as StringUtil from "../../StringUtil";
+import * as Language from "../../Language";
+import User from "../../User";
+import * as DomTraverse from "../../Dom/Traverse";
+import * as Clipboard from "../../Clipboard";
+import * as UiNotification from "../Notification";
+
+/**
+ * Copies one of links to the clipboard.
+ */
+async function copy(event: Event): Promise<void> {
+  event.preventDefault();
+
+  const target = event.currentTarget as HTMLElement;
+  const input = DomTraverse.prevBySel(target, 'input[type="text"]') as HTMLInputElement;
+
+  await Clipboard.copyTextToClipboard(input.value);
+
+  UiNotification.show(Language.get("wcf.global.rss.copy.success"));
+}
+
+/**
+ * Opens the dialog with an anonymous and personalized link after clicking on the RSS feed link.
+ */
+function openDialog(event: Event): void {
+  event.preventDefault();
+
+  const alternative = event.currentTarget as HTMLAnchorElement;
+  const linkWithAccessToken = alternative.href;
+
+  const linkWithoutAccessToken = linkWithAccessToken
+    .replace(`at=${User.userId}-${User.accessToken}&`, "&")
+    .replace(new RegExp(`(\\?|&)at=${User.userId}-${User.accessToken}`), "");
```

Comment:
Can't you use the `URL` class for parsing here as well?

Reply:
No, because this results in encoding issues with the controller (e.g. `index.php?board-feed/&at=`) because it contains a slash in the parameter name (`board-feed/`) which in encoded when using `.searchParams.delete('at')` requiring additional code to then revert that encoding again.

Reply:
I did a cursory check at the URI RFC. URIs don't define a format of the query string (i.e. it does not need to be `application/x-www-form-urlencoded`) and they allow the query string to contain a raw slash. In the URI encoding two URIs that differ in `%2F` and `/` are *not* equivalent, but I'm not sure whether that also applies to `application/x-www-form-urlencoded` encoding. My guess would be that `application/x-www-form-urlencoded` considers these to be equivalent, which would make the result ugly, but not wrong.

I'd like to read up on this in more detail, before considering this to be resolved.

Reply:
Okay, I looked into this:
- The query generally is to be considered opaque from a URI viewpoint.
- The URL class and PHP use x-www-form-urlencoded.
- In x-www-form-urlencoded all names and values are urldecoded before being interpreted.
- WoltLab Suite specifically handles an encoded controller name: https://github.com/WoltLab/WCF/blob/56e3b7959ec8ccd10c87066cdf075103c5fffc05/wcfsetup/install/files/lib/system/request/RouteHandler.class.php#L357

So: The implicit encoding when using `.searchParams.delete('at')` is ugly at most.